meson.build

```diff
@@ -1445,106 +1446,105 @@ endif
 
 
 # macOS features
-macos_sdk_version_py = find_program(join_paths(source_root, 'TOOLS',
-                                    'macos-sdk-version.py'))
-macos_sdk_info = ['', '0.0']
 if darwin
+    macos_sdk_version_py = find_program(join_paths(source_root, 'TOOLS',
+                                        'macos-sdk-version.py'))
     macos_sdk_info = run_command(macos_sdk_version_py, check: true).stdout().split(',')
-endif
 
-macos_sdk_path = macos_sdk_info[0].strip()
-macos_sdk_version = macos_sdk_info[1]
-if macos_sdk_path != ''
-    message('Detected macOS sdk path: ' + macos_sdk_path)
-endif
+    macos_sdk_path = macos_sdk_info[0].strip()
+    macos_sdk_version = macos_sdk_info[1]
+    if macos_sdk_path != ''
+        message('Detected macOS sdk path: ' + macos_sdk_path)
+    endif
 
-if macos_sdk_version != '0.0'
-    message('Detected macOS SDK: ' + macos_sdk_version)
-    add_languages('objc')
-    objc_link_flags = ['-isysroot', macos_sdk_path, '-L/usr/lib', '-L/usr/local/lib']
-    add_project_link_arguments(objc_link_flags, language: ['c', 'objc'])
-endif
+    if macos_sdk_version != '0.0'
+        message('Detected macOS SDK: ' + macos_sdk_version)
+        add_languages('objc')
+        objc_link_flags = ['-isysroot', macos_sdk_path, '-L/usr/lib', '-L/usr/local/lib']
+        add_project_link_arguments(objc_link_flags, language: ['c', 'objc'])
+    endif
 
-macos_10_11_features = get_option('macos-10-11-features').require(
-    macos_sdk_version.version_compare('>=10.11'),
-    error_message: 'a suitable macos sdk version could not be found!',
-)
+    macos_10_11_features = get_option('macos-10-11-features').require(
+        macos_sdk_version.version_compare('>=10.11'),
+        error_message: 'a suitable macos sdk version could not be found!',
+    )
```

Comment:
Good point. I'll drop the darwin part of the PR. Somebody who knows how all the different macos development parts fit together should do this.

Reply:
>Somebody who knows how all the different macos development parts fit together should do this.

AKA me because I wrote it hah. I recall sfan5 complained to me about the sdk always being checked before and I never fixed it for some reason (hope I didn't give some lame excuse because it's not that hard).